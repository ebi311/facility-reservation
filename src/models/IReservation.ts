import { Dayjs } from 'dayjs';
import { ISystem } from './ISystem';

export interface IReservation {
  id: string;
  facilityId: string;
  subject: string;
  description: string;
  startDate: Dayjs;
  endDate: Dayjs;
  system: ISystem;
}
