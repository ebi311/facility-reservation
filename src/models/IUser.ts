export interface IUser {
  displayName: string;
  email: string;
  face?: string;
}
