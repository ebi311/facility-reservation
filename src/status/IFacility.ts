import ISystem from './ISystem';

export default interface IFacility {
  id: string;
  name: string;
  description: string;
  system: ISystem;
}
