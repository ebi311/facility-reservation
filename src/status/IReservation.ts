import ISystem from './ISystem';
import { Dayjs } from 'dayjs';

export default interface IReservation {
  id: string;
  subject: string;
  facilityId: string;
  description?: string;
  startDate: Dayjs;
  endDate: Dayjs;
  system: ISystem;
}
