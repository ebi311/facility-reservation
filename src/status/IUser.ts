export default interface IUser {
  displayName: string;
  faceUrl: string;
}
