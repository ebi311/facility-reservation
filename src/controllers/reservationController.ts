import IReservation from '../status/IReservation';

import superagent from 'superagent';

export const getReservations = async (date: Date): Promise<IReservation[]> => {
  const result = await superagent.get('/api/reservations/').query({
    date: date.toISOString(),
  });
  return result.body as IReservation[];
};

export const getReservationById = async (id: string): Promise<IReservation> => {
  const result = await superagent.get('/api/reservations/' + id);
  return result.body as IReservation;
};
