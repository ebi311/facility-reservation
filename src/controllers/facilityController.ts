import IFacility from '../status/IFacility';
import superagent from 'superagent';
// import URL from 'url';

export const getFacilities = async (): Promise<IFacility[]> => {
  const result = await superagent.get('/api/facilities/').catch(e => {
    throw e;
  });
  return result.body as IFacility[];
};

export const getFacility = async (id: string): Promise<IFacility | null> => {
  const encId = encodeURI(id);
  const result = await superagent.get(`/api/facilities/${encId}`).catch(e => {
    console.error(e);
    throw e;
  });
  return result.body as IFacility;
};

export const postFacility = async (
  data: IFacility,
): Promise<IFacility | null> => {
  const result = await superagent
    .post('/api/facilities/')
    .send(data)
    .catch(e => {
      console.error(e);
      throw e;
    });
  return result.body as IFacility;
};

export const putFacility = async (
  data: IFacility,
): Promise<IFacility | null> => {
  const id = encodeURI(data.id);
  const result = await superagent
    .put('/api/facilities/' + id)
    .send(data)
    .catch(e => {
      console.error(e);
      throw e;
    });
  return result.body as IFacility;
};

export const deleteFacility = async (id: string): Promise<boolean> => {
  await superagent.delete('/api/facilities/' + id).catch(e => {
    console.error(e);
    throw e;
  });
  return true;
};
