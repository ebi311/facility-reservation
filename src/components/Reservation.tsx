import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import Chip from '@material-ui/core/Chip';
import Container from '@material-ui/core/Container';
import FormControl from '@material-ui/core/FormControl';
import Grid from '@material-ui/core/Grid';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Paper from '@material-ui/core/Paper';
import Select from '@material-ui/core/Select';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import DeleteIcon from '@material-ui/icons/Delete';
import DoneIcon from '@material-ui/icons/Done';
import { DateTimePicker } from '@material-ui/pickers';
import dayjs, { Dayjs } from 'dayjs';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { useHistory, useParams } from 'react-router-dom';
import { getFacilities } from '../controllers/facilityController';
import {
  getReservation,
  postReservation,
  putReservation,
} from '../controllers/reservationController';
import { IFacility } from '../models/IFacility';
import { IReservation } from '../models/IReservation';
import { getCurrentUser } from '../auth';

const initReservation = (day: Dayjs): IReservation => {
  const user = getCurrentUser();
  return {
    id: '',
    facilityId: '',
    subject: '',
    description: '',
    startDate: day.startOf('hour'),
    endDate: day.add(1, 'hour').startOf('hour'),
    system: {
      createDate: new Date(),
      createUser: {
        displayName: user?.displayName || '',
        email: user?.email || '',
        face: user?.photoURL || '',
      },
      lastUpdateUser: {
        displayName: user?.displayName || '',
        email: user?.email || '',
        face: user?.photoURL || '',
      },
      lastUpdate: new Date(),
    },
  };
};

const useStyle = makeStyles((theme) => ({
  paper: {
    padding: theme.spacing(1),
    '& > div': {
      marginBottom: theme.spacing(2),
    },
  },
  rightActions: {
    textAlign: 'right',
  },
  cancelButton: {
    color: theme.palette.error.main,
  },
}));

export const Reservation: React.FC = () => {
  const style = useStyle();
  const hour = useMemo(() => {
    const params = new URLSearchParams(window.location.search);
    return dayjs(params.get('date') || undefined);
  }, [window.location.search]);
  const { id } = useParams<{ id: string }>();
  const [reservation, setReservation] = useState(initReservation(hour));
  const [facilities, setFacilities] = useState<IFacility[]>([]);
  const { system } = reservation;
  const { errors, control, reset, trigger, getValues } = useForm<IReservation>({
    defaultValues: reservation,
    mode: 'onBlur',
  });
  useEffect(() => {
    getFacilities().then((result) => {
      setFacilities(result);
    });
    if (!id) return;
    getReservation(id).then((result) => {
      setReservation(result);
      reset(result);
    });
  }, [id]);
  const facilityMenuItems = useMemo(() => {
    return facilities.map((f) => (
      <MenuItem key={f.id} value={f.id}>
        {f.name}
      </MenuItem>
    ));
  }, [facilities]);

  const history = useHistory();
  const onSave = useCallback(async () => {
    const result = await trigger();
    if (!result) return;
    const inputValue = { ...reservation, ...getValues() };
    if (!id) {
      // 新規作成
      const id = await postReservation(inputValue);
      history.replace('/reservation/' + id);
    } else {
      // 更新
      await putReservation(inputValue);
      window.location.reload();
    }
  }, [id, reservation, trigger, getValues]);
  return (
    <Container maxWidth="sm">
      <Paper className={style.paper}>
        <FormControl>
          <InputLabel id="facility-label">設備</InputLabel>
          <Controller
            name="facilityId"
            control={control}
            rules={{ required: true }}
            render={({ value, onChange }) => (
              <Select
                labelId="facility-label"
                value={value}
                onChange={onChange}
              >
                {facilityMenuItems}
              </Select>
            )}
          />
        </FormControl>
        <div style={{ display: 'flex' }}>
          <Controller
            control={control}
            name="startDate"
            render={(data) => {
              return (
                <DateTimePicker
                  value={data.value}
                  onChange={data.onChange}
                  onBlur={data.onBlur}
                  label="開始日時"
                  format="YYYY/MM/DD HH:mm"
                  ampm={false}
                  minutesStep={15}
                />
              );
            }}
          />
          <p>～</p>
          <Controller
            control={control}
            name="endDate"
            render={(data) => {
              return (
                <DateTimePicker
                  value={data.value}
                  onChange={data.onChange}
                  onBlur={data.onBlur}
                  label="終了日時"
                  format="YYYY/MM/DD HH:mm"
                  ampm={false}
                  minutesStep={15}
                />
              );
            }}
          />
        </div>
        <Controller
          control={control}
          name="subject"
          rules={{ required: true }}
          as={
            <TextField
              label="目的"
              fullWidth
              error={!!errors.subject}
              helperText={errors.subject ? '必須です' : ''}
            />
          }
        />
        <Controller
          control={control}
          name="description"
          as={<TextField label="詳細" fullWidth multiline value="" />}
        />
        <InputLabel shrink>登録者</InputLabel>
        <p>
          <Chip
            label={system.createUser.displayName}
            avatar={<Avatar src={system.createUser.face} />}
          />
          {dayjs(system.createDate).format('YYYY-MM-DD HH:mm')}
        </p>
        <InputLabel shrink>更新者</InputLabel>
        <p>
          <Chip
            label={system.lastUpdateUser.displayName}
            avatar={<Avatar src={system.lastUpdateUser.face} />}
          />
          {dayjs(system.lastUpdate).format('YYYY-MM-DD HH:mm')}
        </p>
        <Grid container>
          <Grid item xs={6}>
            <Button className={style.cancelButton} startIcon={<DeleteIcon />}>
              削除
            </Button>
          </Grid>
          <Grid item xs={6} className={style.rightActions}>
            <Button
              variant="contained"
              color="primary"
              startIcon={<DoneIcon />}
              onClick={onSave}
            >
              保存
            </Button>
          </Grid>
        </Grid>
      </Paper>
    </Container>
  );
};
