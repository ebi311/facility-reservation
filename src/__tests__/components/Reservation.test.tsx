/**
 * @jest-environment jsdom
 */
/* eslint-disable @typescript-eslint/no-explicit-any */
import Utils from '@date-io/dayjs';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { fireEvent, render, waitFor } from '@testing-library/react';
import dayjs from 'dayjs';
import MockDate from 'mockdate';
import React from 'react';
import { Reservation } from '../../components/Reservation';
import { getFacilities } from '../../controllers/facilityController';
import { getReservation } from '../../controllers/reservationController';
import { IFacility } from '../../models/IFacility';
import { IReservation } from '../../models/IReservation';
import { IUser } from '../../models/IUser';
import { useParams } from 'react-router-dom';
import { mocked } from 'ts-jest/utils';

MockDate.set(new Date('2021-07-01T11:00:00+0900'));

jest.mock('../../controllers/facilityController', () => ({
  getFacilities: jest.fn().mockImplementation(
    async () =>
      [
        {
          id: 'f001',
          name: '設備001',
          system: {} as any,
        },
        {
          id: 'f002',
          name: '設備002',
          system: {} as any,
        },
      ] as IFacility[],
  ),
}));

jest.mock('../../controllers/reservationController', () => ({
  getReservation: jest.fn(
    async () =>
      ({
        id: 'r001',
        subject: '予約1',
        description: 'r001-説明',
        startDate: dayjs('2021-07-01T10:00:00+0900'),
        endDate: dayjs('2021-07-01T11:00:00+0900'),
        facilityId: 'f001',
        system: {
          createDate: new Date('2021-07-01T08:00:00+0900'),
          createUser: {
            displayName: 'ebihara kenji',
            email: 'ebihara@example.com',
            face: 'http://example.com/face.png',
          },
          lastUpdate: new Date('2021-07-01T09:00:00+0900'),
          lastUpdateUser: {
            displayName: 'ebihara kenji',
            email: 'ebihara@example.com',
            face: 'http://example.com/face.png',
          },
        },
      } as IReservation),
  ),
  postReservation: jest.fn(async () => 'r999'),
  putReservation: jest.fn(async () => ({})),
}));

jest.mock('../../auth', () => ({
  getCurrentUser: jest.fn(
    () =>
      ({
        displayName: 'ebihara kenji',
        email: 'kenji@example.com',
      } as IUser),
  ),
}));

let replace: jest.Mock;

jest.mock(
  'react-router-dom',
  () =>
    ({
      useParams: jest.fn(() => ({ id: '' })),
      useHistory: () => {
        replace = jest.fn();
        return replace;
      },
    } as any),
);

test('新規の画面を開いたとき', async () => {
  const { asFragment, queryByText, getByLabelText } = render(
    <MuiPickersUtilsProvider utils={Utils}>
      <Reservation />
    </MuiPickersUtilsProvider>,
  );
  await waitFor(() => expect(getFacilities).toBeCalledTimes(1));
  const selectFacility = getByLabelText('設備');
  fireEvent.mouseDown(selectFacility);
  expect(!!queryByText('設備001')).toBeTruthy();
  expect(!!queryByText('設備002')).toBeTruthy();
  expect(asFragment()).toMatchSnapshot();
});

test('既存の画面を開いたとき', async () => {
  mocked(useParams).mockReturnValueOnce({ id: 'r001' });
  const { asFragment, getByTestId, getByLabelText } = render(
    <MuiPickersUtilsProvider utils={Utils}>
      <Reservation />
    </MuiPickersUtilsProvider>,
  );
  await waitFor(() => expect(getReservation).toBeCalledTimes(1));
  expect(asFragment()).toMatchSnapshot();
  const selectFacility = getByLabelText('設備') as HTMLDivElement;
  expect(selectFacility.textContent).toBe('設備001');

  const startDate = getByTestId('start-date').querySelector('input');
  expect(startDate?.value).toBe('2021/07/01 10:00');

  const endDate = getByTestId('end-date').querySelector('input');
  expect(endDate?.value).toBe('2021/07/01 11:00');

  const subject = getByTestId('subject').querySelector('input');
  expect(subject?.value).toBe('予約1');

  const description = getByTestId('description').querySelector('textarea');
  expect(description?.value).toBe('r001-説明');

  const create = getByTestId('create');
  expect(create.textContent).toBe('ebihara kenji2021-07-01 08:00');

  const update = getByTestId('update');
  expect(update.textContent).toBe('ebihara kenji2021-07-01 09:00');
});
