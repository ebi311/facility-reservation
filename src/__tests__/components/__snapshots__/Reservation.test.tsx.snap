// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`新規の画面を開いたとき 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="container"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-focused Mui-focused"
          data-shrink="true"
          id="facility-label"
        >
          設備
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-expanded="true"
            aria-haspopup="listbox"
            aria-labelledby="facility-label"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOpen"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        style="display: flex;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
          data-testid="start-date"
          format="YYYY/MM/DD HH:mm"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            開始日時
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              readonly=""
              type="text"
              value="2021/07/01 11:00"
            />
          </div>
        </div>
        <p>
          ～
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root"
          data-testid="end-date"
          format="YYYY/MM/DD HH:mm"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            終了日時
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              readonly=""
              type="text"
              value="2021/07/01 12:00"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="subject"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          目的
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="subject"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="description"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          詳細
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            name="description"
            rows="1"
            style="height: 0px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
            tabindex="-1"
          />
        </div>
      </div>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
      >
        登録者
      </label>
      <div
        data-testid="create"
      >
        <div
          class="MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <span
            class="MuiChip-label"
          >
            ebihara kenji
          </span>
        </div>
        2021-07-01 11:00
      </div>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
      >
        更新者
      </label>
      <div
        data-testid="update"
      >
        <div
          class="MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <span
            class="MuiChip-label"
          >
            ebihara kenji
          </span>
        </div>
        2021-07-01 11:00
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-cancelButton-3"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              削除
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root makeStyles-rightActions-2 MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                  />
                </svg>
              </span>
              保存
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`既存の画面を開いたとき 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root makeStyles-paper-4 MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="container"
    >
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
          id="facility-label"
        >
          設備
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="facility-label"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            role="button"
            tabindex="0"
          >
            設備001
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="f001"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        style="display: flex;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
          data-testid="start-date"
          format="YYYY/MM/DD HH:mm"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            開始日時
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              readonly=""
              type="text"
              value="2021/07/01 10:00"
            />
          </div>
        </div>
        <p>
          ～
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root"
          data-testid="end-date"
          format="YYYY/MM/DD HH:mm"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            終了日時
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              readonly=""
              type="text"
              value="2021/07/01 11:00"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="subject"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          目的
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="subject"
            type="text"
            value="予約1"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="description"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          詳細
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            name="description"
            rows="1"
            style="height: 0px; overflow: hidden;"
          >
            r001-説明
          </textarea>
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
            tabindex="-1"
          />
        </div>
      </div>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
      >
        登録者
      </label>
      <div
        data-testid="create"
      >
        <div
          class="MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar"
          >
            <img
              class="MuiAvatar-img"
              src="http://example.com/face.png"
            />
          </div>
          <span
            class="MuiChip-label"
          >
            ebihara kenji
          </span>
        </div>
        2021-07-01 08:00
      </div>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
      >
        更新者
      </label>
      <div
        data-testid="update"
      >
        <div
          class="MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar"
          >
            <img
              class="MuiAvatar-img"
              src="http://example.com/face.png"
            />
          </div>
          <span
            class="MuiChip-label"
          >
            ebihara kenji
          </span>
        </div>
        2021-07-01 09:00
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-cancelButton-6"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              削除
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root makeStyles-rightActions-5 MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                  />
                </svg>
              </span>
              保存
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
